baselines/depthfl/README.md

```diff
@@ -0,0 +1,170 @@
+python -m depthfl.main --config-name="heterofl" # HeteroFL
+python -m depthfl.main --config-name="heterofl" exclusive_learning=true model_size=1 # exclusive learning - 100% (a)
+```
+
+### Stateful clients comment
+
+To implement `feddyn`, stateful clients that store prev_grads information are needed. Since flwr does not yet officially support stateful clients, it was implemented as a temporary measure by loading `prev_grads` from disk when creating a client, and then storing it again on disk after learning. Specifically, there are files that store the state of each client in the `prev_grads` folder. When the strategy is instantiated (for both `FedDyn` and `HeteroFL`) the content of `prev_grads` is reset. 
+
+
+## Expected Results
+
+With the following command we run DepthFL (FedDyn / FedAvg), InclusiveFL, and HeteroFL to replicate the results of table 2,3,4 in DepthFL paper. Tables 2, 3, and 4 may contain results from the same experiment in multiple tables. 
+
+```bash
+# table 2 (HeteroFL row)
+python -m depthfl.main --config-name="heterofl" 
+python -m depthfl.main --config-name="heterofl" --multirun exclusive_learning=true model.scale=false model_size=1,2,3,4 
+
+# table 2 (DepthFL(FedAvg) row)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false 
+python -m depthfl.main --multirun fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=1,2,3,4
+
+# table 2 (DepthFL row)
+python -m depthfl.main
+python -m depthfl.main --multirun exclusive_learning=true model_size=1,2,3,4
+```
+
+**Table 2** 
+
+100% (a), 75%(b), 50%(c), 25% (d) cases are exclusive learning scenario. 100% (a) exclusive learning means, the global model and every local model are equal to the smallest local model, and 100% clients participate in learning. Likewise, 25% (d) exclusive learning means, the global model and every local model are equal to the larget local model, and only 25% clients participate in learning.
+
+| Scaling Method | Dataset | Global Model | 100% (a) | 75% (b) | 50% (c) | 25% (d) | 
+| :---: | :---: | :---: | :---: | :---: | :---: | :---: | 
+| HeteroFL<br>DepthFL (FedAvg)<br>DepthFL | CIFAR100 | 57.61<br>72.67<br>76.06 | 64.39<br>67.08<br>69.68 | 66.08<br>70.78<br>73.21 | 62.03<br>68.41<br>70.29 | 51.99<br>59.17<br>60.32 |
+
+```bash
+# table 3 (Width Scaling - Duplicate results from table 2)
+python -m depthfl.main --config-name="heterofl" 
+python -m depthfl.main --config-name="heterofl" --multirun exclusive_learning=true model.scale=false model_size=1,2,3,4 
+
+# table 3 (Depth Scaling : Exclusive Learning, DepthFL(FedAvg) rows - Duplicate results from table 2)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false 
+python -m depthfl.main --multirun fit_config.feddyn=false fit_config.kd=false  exclusive_learning=true model_size=1,2,3,4
+
+## table 3 (Depth Scaling - InclusiveFL row)
+python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false
```

Comment:
```suggestion
python -m depthfl.main --multirun fit_config.feddyn=false fit_config.kd=false fit_config.extended=false model_size=1,2,3,4
```

Reply:
I believe we need to add the `--mutlirun` and `model_size=1,2,3,4`? this is equivalent to doing:
```
python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false model_size=1
python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false model_size=2
python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false model_size=3
python -m depthfl.main fit_config.feddyn=false fit_config.kd=false fit_config.extended=false model_size=4
```

Reply:
There is no need to add multirun. You can get all the results of the row in one execution.